### ISSUE TITLE
s.options.numOutputBusChannels defaults to 2 in SC >= 3.7.0

### ISSUE THREAD

**Issue by joslloand** [ambisonictoolkit/atk-sc3#19]
This noted change, [s.options.numOutputBusChannels defaults to 2 in SC >= 3.7.0](http://new-supercollider-mailing-lists-forums-use-these.2681727.n2.nabble.com/s-options-numOutputBusChannels-defaults-to-2-in-SC-gt-3-7-0-td7626073.html), "breaks" ATK examples.

Consider adding a note to [Introducing the Ambisonic Toolkit](http://doc.sccode.org/Guides/Intro-to-the-ATK.html) stating server `numOutputBusChannels` should be set to desired number of outputs. In the least, `numOutputBusChannels` should be set to [`decoder.numChannels`](http://doc.sccode.org/Classes/FoaDecoderMatrix.html#-numChannels).

We might consider this a _critical_ issue for new users.

**Comment by joslloand** [comment]
@dmartinp, we might as well consider this as related to the clarifications & revisions you're addressing.

**Comment by dmartinp** [comment]
Include simple check of server output channels to see if it is greater than or equal to the number decoder channels (realtime only).

**Comment by dmartinp** [comment]
See commit [17a399c857cb8bd63e97c9e7ece4ffe944244591](https://github.com/ambisonictoolkit/atk-sc3/commit/17a399c857cb8bd63e97c9e7ece4ffe944244591).  This could be an ok check, the only problem comes when using NRT or a server other than the default server.  As you can see now, I check the default server (is there a way to know which server is being used?) for the numOutputBusChannels and check if it is less than the number of outputs for each decoder.  We can either stick with this, or provide warnings in the documentation, or both.  Have not worked on putting a warning in the help files.

**Comment by joslloand** [comment]
Need to add some information to Help about how to set number of output channels. So this could be in the Intro-to-the-ATK and/or in the workshop files.

**Comment by joslloand** [comment]
Question: do we also need to do something similar to check for a valid number of input channels?

**Comment by joslloand** [comment]
Actually, @dmartinp, after looking at this more closely, I think the warning raised by https://github.com/ambisonictoolkit/atk-sc3/commit/17a399c857cb8bd63e97c9e7ece4ffe944244591 isn't actually what we want to do.

Imagine that you make a synthDef that processes by decoding to a t-design of 1000 virtual speakers, processes each somehow, and then re-encodes, all in the same synthDef. When you make that decoder, you're not going to want to see an error thrown, as you aren't doing anything irrational.

This is really a SuperCollider issue, rather than an ATK issue.

Another use case: Decode to 1000 virtual speakers, send to a private bus, re-encode from that bus, then output to a small number of outs (quad maybe?). In this form, we'd get a warning, as well.

Another SuperCollider 'inbuilt' UGen use case: [`PanAz.ar`](http://doc.sccode.org/Classes/PanAz.html) with `numChans = 1000`. No error is thrown if the user attempts to send to outputs that don't exist.

Yeah... so I'm thinking this is really just a documentation issue that is best resolved by including example code something like :

```
// check number of decoder output channels against server
(myDecoder.numChannels > myServer.numOutputBusChannels).if({
    "Number of output bus channels is less than the number of outputs from the decoder. Please set the output bus channels to: %".format(myDecoder.numChannels).warn
})
```

**Comment by joslloand** [comment]
Reverted 17a399c857cb8bd63e97c9e7ece4ffe944244591

**Comment by joslloand** [comment]
Relevant docs:
- [Introducing the Ambisonic Toolkit](http://doc.sccode.org/Guides/Intro-to-the-ATK.html#Choose a decoder)
- [FoaDecode](http://doc.sccode.org/Classes/FoaDecode.html#examples)
- [FoaEncode](http://doc.sccode.org/Classes/FoaEncode.html#examples)
- [FoaTransform](http://doc.sccode.org/Classes/FoaTransform.html#examples)
- [FoaXform](http://doc.sccode.org/Classes/FoaXform.html#examples)

It could be that just a single note or section in [Introducing the Ambisonic Toolkit](http://doc.sccode.org/Guides/Intro-to-the-ATK.html#Choose a decoder) would be appropriate, with links to that section.

**Comment by joslloand** [comment]
Closed with fe86e3212d9d5b24715efbd0181ff3882a531e9b
